builder/resubmit_utils.go

```diff
@@ -9,13 +9,28 @@ import (
 )
 
 // runResubmitLoop checks for update signal and calls submit respecting provided rate limiter and context
-func runResubmitLoop(ctx context.Context, limiter *rate.Limiter, updateSignal chan struct{}, submit func()) {
+func runResubmitLoop(ctx context.Context, limiter *rate.Limiter, updateSignal chan struct{}, submit func(), submitTime *time.Time) {
 	for {
 		select {
 		case <-ctx.Done():
 			return
 		case <-updateSignal:
-			res := limiter.Reserve()
+			var res *rate.Reservation
+			if submitTime == nil {
+				res = limiter.Reserve()
+			} else {
+				now := time.Now()
+				sleepTime := submitTime.UTC().UnixMilli() - now.UTC().UnixMilli()
```

Comment:
> why this instead of just time.Sub without converting to millis and back (in sleep)

I was debugging timestamp issue because input `attrs.Timestamp` was getting set to the year `1970` so `time.Sub` was incorrect. But that's fixed now so I'll update 👍🏾

Reply:
updated